docs/read-from-object-store-layer.md

```diff
@@ -0,0 +1,141 @@
+   ```
+
+2. Initialize a TypeScript project.
+
+   ```shell
+   tsc --init
+   ```
+
+3. Install node types.
+
+   ```shell
+   npm install --save-dev @types/node
+   ```
+
+4. Install the SDK modules.
+
+   ```shell
+   npm install --save @here/olp-sdk-authentication @here/olp-sdk-dataservice-api
+   ```
+
+   Now, everything is set to create the app.
+
+5. Create the `index.ts` file and app skeleton.
+
+   ```typescript
+   /**
+    * An example of the Node.js app used for reading data from and writing it to an object store layer in the datastore.
+    */
+
+   class App {
+     run() {
+       console.log("App works!");
+     }
+   }
+
+   const app = new App();
+   app.run();
+   ```
+
+6. Compile and run the app.
+
+   ```shell
+   tsc && node .
+   ```
+
+After a successful run, the console displays the following message:
+
+```shell
+App works!
+```
+
+## Use `ObjectStoreApi` from `@here/olp-sdk-dataservice-api`
+
+You can use the `ObjectStoreApi` functions to request any data from an [object store layer](https://developer.here.com/documentation/data-user-guide/user_guide/portal/layers/layers.html#object-store-layers).
+
+**To use the `ObjectStoreApi` functions:**
+
+1. Create the `OlpClientSettings` object.
+
+   For instructions, see <a href="https://github.com/heremaps/here-data-sdk-typescript/blob/master/docs/create-platform-client-settings.md" target="_blank">Create platform client settings</a>.
+
+2. Create a `RequestBuilder` instance with `RequestFactory` that contains the catalog HRN, the platform client settings from step 1, API name, and API version.
+
+   ```typescript
+   const requestBuilder = await RequestFactory.create(
+      "blob", 
+      "v2", 
+      settings, 
+      HRN.fromString("your-catalog-hrn")
+   );
+   ```
+
+## Get data from an object store layer
+
+You can request any data from an [object store layer](https://developer.here.com/documentation/data-user-guide/user_guide/portal/layers/layers.html#object-store-layers) using a key or datahandle.
+
+**To get data from the object store layer:**
+
+1. Create the `RequestBuilder` object.
+
+   For instructions, see [Create RequestBuilder](#use-objectstoreapi-from-hereolp-sdk-dataservice-api).
+
+2. Call the `getBlobByKey` function with the `key` and `layerId` parameters.
+
+   ```typescript
+   const result = await ObjectStoreApi.getBlobByKey(requestBuilder, {
+      key: "your-data-key",
+      layerId: "your-layer-id"
+   });
+   ```
+
+You receive data from the requested key of the selected layer.
+
+## Get all keys
+
+1. Create the `RequestBuilder` object.
+
+   For instructions, see [Create RequestBuilder](#use-objectstoreapi-from-hereolp-sdk-dataservice-api).
+
+2. Call the `listKeys` function with the `layerId` parameter.
+
+   ```typescript
+   const result = await ObjectStoreApi.listKeys(requestBuilder, {
+      layerId: "your-layer-id"
+   });
+   ```
+You receive a list of all keys from the selected layer.
```

Comment:
It should be noted that the listkeys returns only one level of keys. The response will contain for each key an indication if it is a folder or object. For each folder, the client must recursively call this end point with a parent in the query string to retrieve the next level of keys for the folder.